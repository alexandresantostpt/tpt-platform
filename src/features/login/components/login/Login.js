import PropTypes from 'prop-types'
import React from 'react'

import Background from '../background/Background'
import Box from '../box/Box'
import Page from '../page/Page'

const Login = ({ children }) => (
    <Page>
        <Background />
        <Box>{children}</Box>
    </Page>
)

Login.propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.func, PropTypes.object, PropTypes.string]).isRequired
}

export default Login
