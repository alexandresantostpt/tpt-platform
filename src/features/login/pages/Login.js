import PropTypes from 'prop-types'
import React, { useState } from 'react'

import { i18n } from '@i18n'

import Form from '../components/form/Form'
import Button from '../components/button/Button'
import Text from '../components/text/Text'
import Input from '../components/input/Input'
import ErrorMessage from '../components/errorMessage/ErrorMessage'
import LinkTo from '../components/link/LinkTo'
import LoginComponent from '../components/login/Login'

import { routesForgotPassword } from '../routes'

const Login = props => {
    const { wrongCredentials } = props

    const [login, updateLogin] = useState('')
    const [password, updatePassword] = useState('')

    const loginButtonClicked = event => {
        event.preventDefault()
        const { requestAuthAction } = props

        if (login && password) {
            requestAuthAction({
                password: password,
                user: login
            })
        }
    }

    return (
        <LoginComponent>
            <Text>
                <div>{i18n.t('titles.welcome')}</div>
                <div>{i18n.t('titles.signInToContinue')}</div>
            </Text>
            <Form onSubmit={loginButtonClicked}>
                <ErrorMessage>{wrongCredentials ? i18n.t('messages.errors.auth') : ''}</ErrorMessage>
                <Input id={'login'} onChange={updateLogin} placeholder={i18n.t('placeholders.login')} type="text" />
                <Input id={'password'} onChange={updatePassword} placeholder={i18n.t('placeholders.password')} type="password" />
                <Button disabled={false} text={i18n.t('buttons.login')} type="submit" />
                <LinkTo text={i18n.t('link.forgotPassword')} to={routesForgotPassword.path} />
            </Form>
        </LoginComponent>
    )
}

Login.propTypes = {
    requestAuthAction: PropTypes.func.isRequired,
    wrongCredentials: PropTypes.bool.isRequired
}

export default Login
