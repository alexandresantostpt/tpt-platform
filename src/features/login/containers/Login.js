import React from 'react'

import { connect } from 'react-redux'

import { requestAuthAction } from '../actions'

import GlobalStyle from '@components/globalStyle/GlobalStyle'
import Toasts from '@components/toast/Toasts'

import Login from '../pages/Login'

import { getUserSelector, getWrongCredentialsSelector } from '../selectors'

const LoginContainer = props => (
    <>
        <GlobalStyle />
        <Toasts />
        <Login {...props} />
    </>
)

const mapStateToProps = state => ({
    user: getUserSelector(state),
    wrongCredentials: getWrongCredentialsSelector(state)
})

const mapDispatchToProps = {
    requestAuthAction
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(LoginContainer)
