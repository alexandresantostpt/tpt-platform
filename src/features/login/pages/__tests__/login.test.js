import React from 'react'
import renderer from 'react-test-renderer'

import Login from '../Login'

describe('Tests for Login component', () => {
    it('Should render withou crash', () => {
        const tree = renderer.create(<Login requestAuthAction={jest.fn()} wrongCredentials={false} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
