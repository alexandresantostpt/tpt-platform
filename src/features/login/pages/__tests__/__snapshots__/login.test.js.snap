// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Login component Should render withou crash 1`] = `
<main
  className="Page__StyledPage-sc-9alblw-0 eBjaLI"
>
  <div
    className="Background__StyledBg-sc-904t04-0 bTqgFQ"
  />
  <div
    className="Box__StyledBox-v5mj6b-0 llPnpv"
  >
    <div
      className="Text__StyledText-fa81pq-0 bIShHK"
    >
      <div>
        Bem-vindo!
      </div>
      <div>
        Fa√ßa login para continuar
      </div>
    </div>
    <form
      className="Form__StyledForm-sc-2b0hvf-0 hgpEDk"
      onSubmit={[Function]}
    >
      <div
        className="ErrorMessage__StyledDiv-maht7p-0 faewLc"
      >
        
      </div>
      <div
        className="Input__StyledBox-sc-4bhgmp-0 gcStMN"
      >
        <input
          className="Input__StyledInput-sc-4bhgmp-3 hoXjZP"
          id="login"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          className="Input__StyledLabel-sc-4bhgmp-2 lcosGj"
          htmlFor="login"
        >
           
          Login
           
        </label>
      </div>
      <div
        className="Input__StyledBox-sc-4bhgmp-0 gcStMN"
      >
        <input
          className="Input__StyledInput-sc-4bhgmp-3 kyZuuN"
          id="password"
          onChange={[Function]}
          required={true}
          type="password"
        />
        <label
          className="Input__StyledLabel-sc-4bhgmp-2 lcosGj"
          htmlFor="password"
        >
           
          Senha
           
        </label>
        <button
          className="Input__StyledButton-sc-4bhgmp-1 jreoz"
          onClick={[Function]}
          type="button"
        >
          mostrar
        </button>
      </div>
      <button
        className="Button__StyledButton-sc-6tss9n-0 dHMzfC"
        disabled={false}
        type="submit"
      >
        Entrar
      </button>
      <a
        className="LinkTo__StyledLink-sc-1u70co-0 khxOGt"
        to="/forgotPassword"
      >
        Esqueceu a senha?
      </a>
    </form>
  </div>
</main>
`;
